pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -193,4 +211,29 @@ protected void removeSegmentFromStore(String tableName, String segmentId) {
       LOGGER.info("localDiskDir is not configured, won't delete segment {} from disk", segmentId);
     }
   }
+
+  protected void removeSegmentsFromDeletedDirectory(int deleteRetention) {
+    if (_localDiskDir != null) {
+      File targetDir = new File(_localDiskDir, DELETED_SEGMENTS);
+      // Remove files whose age is older than delete retention. It looks for subdirectories.
+      AgeFileFilter fileFilter = new AgeFileFilter(DateTime.now().minusDays(deleteRetention).toDate());
+      Collection<File> targetFiles = FileUtils.listFiles(targetDir, fileFilter, TrueFileFilter.INSTANCE);
+      for (File f : targetFiles) {
+        if (!f.delete()) {
+          LOGGER.warn("Cannot remove file {} from deleted directory.", f.getAbsolutePath());
+        }
+      }
+
+      // Remove empty directories. We don't need to look for subdirectories for this case.
+      for (File f : targetDir.listFiles((FileFilter) DirectoryFileFilter.DIRECTORY)) {
+        if (f.list().length == 0) {
+          if (!f.delete()) {
+            LOGGER.warn("Cannot remove empty directory {} from deleted directory.", f.getAbsolutePath());
+          }
+        }
+      }
+    } else {
+      LOGGER.info("localDiskDir is not configured, won't delete any expired segments from deleted directory.");
```

Comment:
This is not an error. If localdiskdir is not configured, then sgement files are never moved into Deleted_Segments directory, so logging an info is fine here.